Rakefile

```diff
@@ -4,3 +4,52 @@ require "rspec/core/rake_task"
 RSpec::Core::RakeTask.new(:spec)
 
 task :default => :spec
+
+namespace :openapi do
+  IBM_CLOUD_API_GEMS = %w[ibm_cloud_power ibm_cloud_vpc ibm_cloud_iam ibm_cloud_resource_controller]
+  IBM_CLOUD_OPENAPI_JSON = {
+    "ibm_cloud_power"               => "power-cloud.json",
+    "ibm_cloud_vpc"                 => "vpc.json",
+    "ibm_cloud_iam"                 => "iam-identity-token-api.json",
+    "ibm_cloud_resource_controller" => "resource-controller/resource-controller.json"
+  }
+
+  task :download_cli, [:version] do |_t, args|
+    require "net/http"
+
+    version = args[:version] || "5.0.0-beta2"
+
+    puts "Downloading openapi-generator-cli-#{version}.jar..."
+
+    uri = URI("https://repo1.maven.org/maven2/org/openapitools/openapi-generator-cli/#{version}/openapi-generator-cli-#{version}.jar")
+
+    response = Net::HTTP.get_response(uri)
+    response.value
+
+    puts "Downloading openapi-generator-cli-#{version}.jar...Complete"
+
+    File.write("openapi-generator-cli", response.body)
+  end
+
+  task :download_openapi_specs do
+    IBM_CLOUD_API_GEMS.each do |api_gem|
+      openapi_json = IBM_CLOUD_OPENAPI_JSON[api_gem]
+      url          = "https://cloud.ibm.com/apidocs/#{openapi_json}"
+      output_file  = openapi_json.sub("/", "-")
+
+      puts "Downloading #{url}..."
+      system("curl #{url} > #{output_file} 2>/dev/null")
```

Comment:
Assuming you'll want to Net::HTTP this eventually

Reply:
Yes, this is really weird actually, I can't `wget` the file and I can't `Net::HTTP` it either.  When I run curl in debug mode it is using `HTTP/2` to pull the file down.  I tried with `net-http2` gem but was getting a socket error so I'm going to come back to it later, `system(curl)` works well enough for now

Reply:
WAT

Reply:
@Fryguy found that cloud.ibm.com is blacklisting "ruby" and "wget" by user-agent headers